src/lib.rs

```diff
@@ -1616,62 +1615,64 @@ impl Visibility {
 
 type PhantomUnsendUnsync = PhantomData<*mut ()>;
 
-static AVAILABLE: AtomicBool = AtomicBool::new(true);
-
 /// An empty type which prevents the use of this library from multiple threads simultaneously.
 #[derive(Debug)]
-pub struct Clang(PhantomUnsendUnsync);
+#[non_exhaustive]
+pub struct Clang {
+    #[cfg(feature = "runtime")]
+    libclang: Option<std::sync::Arc<clang_sys::SharedLibrary>>,
+    unsend_unsync: PhantomUnsendUnsync,
+}
 
 impl Clang {
     //- Constructors -----------------------------
 
     /// Constructs a new `Clang`.
     ///
-    /// Only one instance of `Clang` is allowed at a time.
+    /// Only one instance of `Clang` is allowed per thread.
     ///
     /// # Failures
     ///
-    /// * an instance of `Clang` already exists
     /// * a `libclang` shared library could not be found
     /// * a `libclang` shared library symbol could not be loaded
-    #[cfg(feature="runtime")]
+    #[cfg(feature = "runtime")]
     pub fn new() -> Result<Clang, String> {
-        if AVAILABLE.swap(false, atomic::Ordering::SeqCst) {
-            load().map(|_| Clang(PhantomData))
-        } else {
-            Err("an instance of `Clang` already exists".into())
+        if !clang_sys::is_loaded() {
+            clang_sys::load()?;
         }
+
+        let library = clang_sys::get_library().unwrap();
+        Ok(Clang {
+            libclang: Some(library),
+            unsend_unsync: PhantomUnsendUnsync,
+        })
     }
 
     /// Constructs a new `Clang`.
     ///
-    /// Only one instance of `Clang` is allowed at a time.
+    /// Only one instance of `Clang` is allowed per thread.
```

Comment:
Note that I'm not actually sure if this true. Also, this is currently not checked when not using the `runtime` feature. In either case, the current implementation is incorrect. 

Depending on whether `libclang` is thread-safe or not, we either need to allow 

1 `Clang` instance per thread where `Clang: !Send + !Sync`

or we need to allow 

1 `Clang` instance per process where `Clang: Send`.